pkg/pillar/cmd/zedagent/handlenetworkinstance.go

```diff
@@ -81,9 +81,22 @@ func prepareAndPublishNetworkInstanceInfoMsg(ctx *zedagentContext,
 	if !deleted {
 		info.Displayname = status.DisplayName
 		info.InstType = uint32(status.Type)
-		info.CurrentUplinkIntf = status.SelectedUplinkIntfName
+		info.Ports = status.Ports
+		if len(status.Ports) > 0 {
+			// For backward-compatibility with older controllers.
+			// Just report the first port from the list.
```

Comment:
When I read the code I didn't realize that it reports all ports using info.Ports and one for backwards compatibility using the old CurrentUplinkIntf.
How about moving the comment above where you set info.Ports and explaining that it is for new controllers and CurrentUplinkIntf for old in one comment?

Reply:
Done